# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/parent/datamodel/FileSystemFileManager.cpp
# Commit: d63dac1ee4d4
# Full Hash: d63dac1ee4d471defff484dae5a3535e1f8ba072
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-10 22:03:45
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Make it possible to initialize temporary origins without ensuring origin directories; r=dom-storage-reviewers,jari
#   
#   LSNG already uses some QuotaManager APIs to achieve that origin directories are
#   not created if they don't exist during datastore preparation, but the feature
#   is not easy to use and it's also not generalized enough for use in other quota
# ==============================================================================

diff -r 2ab08c49f2b1 -r d63dac1ee4d4 dom/fs/parent/datamodel/FileSystemFileManager.cpp
--- a/dom/fs/parent/datamodel/FileSystemFileManager.cpp	Thu Oct 10 12:22:54 2024 +0000
+++ b/dom/fs/parent/datamodel/FileSystemFileManager.cpp	Thu Oct 10 12:22:55 2024 +0000
@@ -170,10 +170,11 @@
   quota::QuotaManager* quotaManager = quota::QuotaManager::Get();
   MOZ_ASSERT(quotaManager);
 
-  QM_TRY_INSPECT(
-      const auto& fileSystemDirectory,
-      quotaManager->EnsureTemporaryOriginIsInitializedInternal(aOriginMetadata)
-          .map([](const auto& aPair) { return aPair.first; }));
+  QM_TRY_INSPECT(const auto& fileSystemDirectory,
+                 quotaManager
+                     ->EnsureTemporaryOriginIsInitializedInternal(
+                         aOriginMetadata, /* aCreateIfNonExistent */ true)
+                     .map([](const auto& aPair) { return aPair.first; }));
 
   QM_TRY(QM_TO_RESULT(fileSystemDirectory->AppendRelativePath(
       NS_LITERAL_STRING_FROM_CSTRING(FILESYSTEM_DIRECTORY_NAME))));